terraform/modules/platform/variables.tf

```diff
@@ -0,0 +1,33 @@
+variable "app" {
+  description = "The short name for the delivery team or ADO."
+  type        = string
+  validation {
+    condition     = contains(["ab2d", "bcda", "dpc"], var.app)
+    error_message = "Invalid short var.app (application). Must be one of ab2d, bcda, or dpc."
+  }
+}
+
+variable "env" {
+  description = "The solution's environment name."
+  type        = string
+  validation {
+    condition     = one([for x in ["test", "dev", "sandbox", "prod"] : x if can(regex("^${x}$$|^([a-z0-9]+[a-z0-9-])+([^--])-${x}$$", var.env))]) != null
+    error_message = "Invalid environment/workspace name. Must end in one of test, dev, sandbox, or prod."
+  }
+}
+
+variable "service" {
+  description = "Service _or_ terraservice name."
+  type        = string
+}
+
+variable "additional_tags" {
+  default     = {}
+  description = "Additional tags to merge into final default_tags output"
+  type        = map(string)
+}
+
+variable "module_root" {
+  description = "The full URL to the terraform module root at issue for this infrastructure"
+  type        = string
+}
```

Comment:
Do you prefer module_root over root_module? I see it referred to as "root module" in the terraform docs.

Reply:
I'm ambivalent.